### ISSUE TITLE
Bug: `labels.labelDay` does not get called

### ISSUE THREAD

**Issue by jordanoverbye** [gpbl/react-day-picker#1774]
### Bug description

When using the `labels` prop, the value returned from the `labelDay` function never gets used.

### To reproduce

[Minimal Codesandbox](https://codesandbox.io/s/infallible-gould-h6swt8?file=/src/App.tsx)

When you inspect the DOM of the calendar, you'll see there is no aria-label on the table cell or inner button.

#### Sample code

```tsx
import { Labels, DayPicker } from "react-day-picker";
import { format } from "date-fns";

const calendarLabels: Partial<Labels> = {
  labelDay: (day, _, options) => {
    // NOTE: This is never called
    return format(day, "MMMM yyyy (EEEE)", options);
  }
};

export default function App() {
  return <DayPicker mode="single" labels={calendarLabels} />;
}
```

### Expected behavior

The button inside of the cell has an aria-label attribute with the value returned from `labelDay` function

### Screenshots

N/A

### Additional context

Tested in version 8.6.0.

**Comment by vvanmol** [comment]
Hello, 

Is there any progress on this issue ?  I just migrated our application to use the latest version of the DayPicker but not having the ARIA labels attached to the calendar days is a blocking issue on our side ...

**Comment by gpbl** [comment]
@vvanmol in my understanding the aria-label is not required anymore since we moved the calendar to a `grid` role in https://github.com/gpbl/react-day-picker/releases/tag/v8.5.0

If the upgrade is breaking your test, consider the label is now the date of the day.

Some code replicating your issue?

**Comment by dleetr** [comment]
While the grid role upgrade was helpful this is the log of what is read off by NVDA in single mode

![image](https://github.com/gpbl/react-day-picker/assets/55865217/0fb9f2b4-3e8a-40b7-a283-bc612f78183b)

As you can see a user relying on aria labels for navigation when selecting has to recall their month as it's only read off a single time. I don't really see a need to deprecate the function. Maybe I don't want the whole date, maybe I don't want the day of the week read off. It's much more convenient to have this granular control. 
Also, if it's not being used it's beyond deprecated - it's not supported. If you really want to go forward with removing the interface I'd suggest to not offer it as a deprecated field since that gives the expectation you can still use it, just that you won't be able to use it much longer if you keep your deps updated.

**Comment by gpbl** [comment]
Yeah – I agree that deprecating that function was done wrong. It is a bit difficult to get it back now. We will address this in the next major version.

Meanwhile, you can restore the previous usage of `labelDay` or playing with the label of the day content by passing a custom `DayContent` component:

https://codesandbox.io/s/react-daypicker-forked-3hqj25?file=/src/App.tsx

```tsx
import React from "react";
import { DayContentProps, DayPicker, useDayPicker } from "react-day-picker";
import "react-day-picker/dist/style.css";

function DayContent(props: DayContentProps): JSX.Element {
  const {
    locale,
    classNames,
    styles,
    labels: { labelDay },
    formatters: { formatDay }
  } = useDayPicker();

  return (
    <>
      <span aria-hidden="true">{formatDay(props.date, { locale })}</span>
      <span className={classNames.vhidden} style={styles.vhidden}>
        {labelDay(props.date, props.activeModifiers, { locale })}
      </span>
    </>
  );
}

export default function App() {
  const [selected, setSelected] = React.useState<Date>();

  return (
    <div className="App">
      <DayPicker
        mode="single"
        selected={selected}
        onSelect={setSelected}
        labels={{ labelDay: (day) => `=${day.getDate().toString()}` }}
        components={{ DayContent }}
      />
    </div>
  );
}
```
